apache-http-client/src/main/java/com/joyent/http/signature/apache/httpclient/HttpSignatureCache.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.http.signature.apache.httpclient;
+
+import com.joyent.http.signature.HttpSignatureException;
+import com.joyent.http.signature.Signer;
+import org.apache.http.auth.AuthenticationException;
+import org.apache.http.auth.Credentials;
+
+import java.security.KeyPair;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Class that an interface to caching HTTP signatures for a given
+ * {@link Credentials} instance.
+ *
+ * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
+ * @since 3.0.0
+ */
+class HttpSignatureCache {
+    /**
+     * Credentials to associate cache with.
+     */
+    private final Credentials credentials;
+
+    /**
+     * Reference to the date time of the last HTTP signature.
+     */
+    private final AtomicReference<String> lastDate = new AtomicReference<>("");
```

Comment:
Per experiments, I don't think there is doing anything more complicated than a simple synchronized bock to protect these fields.  If References are used there should be a comment explaining how they interact with the synchronized blocks.